variables.tf

```diff
@@ -886,53 +932,65 @@ variable "role_assignments" {
   nullable    = false
 }
 
-variable "role_based_access_control_enabled" {
-  type        = bool
-  default     = true
-  description = "Whether or not role-based access control is enabled for the Kubernetes cluster."
-}
-
-variable "run_command_enabled" {
-  type        = bool
-  default     = false
-  description = "Whether or not the run command is enabled for the Kubernetes cluster."
-}
-
 variable "service_mesh_profile" {
   type = object({
-    mode                             = string
-    internal_ingress_gateway_enabled = optional(bool)
-    external_ingress_gateway_enabled = optional(bool)
-    revisions                        = optional(list(string), [])
-    certificate_authority = optional(object({
-      key_vault_id           = string
-      root_cert_object_name  = string
-      cert_chain_object_name = string
-      cert_object_name       = string
-      key_object_name        = string
+    mode = string
+    istio = optional(object({
+      certificateAuthority = optional(object({
+        plugin = optional(object({
+          certChainObjectName = optional(string)
+          certObjectName      = optional(string)
+          keyObjectName       = optional(string)
+          keyVaultId          = optional(string)
+          rootCertObjectName  = optional(string)
+        }))
+      }))
+      components = optional(object({
+        egressGateways = optional(object({
+          enabled                  = bool
+          gatewayConfigurationName = optional(string)
+          name                     = optional(string)
+          namespace                = optional(string)
+        }))
+        ingressGateways = optional(object({
+          enabled = bool
+          mode    = optional(string)
+        }))
```

Comment:
When testing this ingressGateway configuration with an Automatic Private Cluster I get the following error when trying to provision the cluster.

"message": "Invalid request body. Converting request body to a managed cluster encountered error: IstioComponents.properties.serviceMeshProfile.istio.components.ingressGateways accept type []*v20251001.IstioIngressGateway, not type object.",

I did wonder why this design would drop support for a cluster having both internal and external ingress gateways which is a supported design.